test/configuration_test.dart

```diff
@@ -121,15 +122,35 @@ Future<void> main([List<String>? args]) async {
     Configuration config = Configuration([Car.schema], inMemory: true);
     var realm = Realm(config);
 
-    config.isReadOnly = true;
-    expect(() => Realm(config), throws<RealmException>("Realm at path '${config.path}' already opened with different read permissions"));
+    expect(() {
+      config = Configuration([Car.schema]);
+      config.isReadOnly = true;
+      Realm(config);
+    }, throws<RealmException>("Realm at path '${config.path}' already opened with different read permissions"));
+
     realm.close();
   });
-  
+
   test('Configuration - FIFO files fallback path', () {
     Configuration config = Configuration([Car.schema], fifoFilesFallbackPath: "./fifo_folder");
     var realm = Realm(config);
     realm.close();
   });
-  
+
+  test('Configuration.operator== equal configs', () {
+    final config = Configuration([Dog.schema, Person.schema]);
+    final realm = Realm(config);
+    expect(config, realm.config);
+    realm.close();
+  });
+ 
+  test('Configuration.operator== different configs', () {
+    var config = Configuration([Dog.schema, Person.schema]);
+    final realm1 = Realm(config);
+    config = Configuration([Dog.schema, Person.schema]);
+    final realm2 = Realm(config);
+    expect(realm1.config, isNot(realm2.config));
+    realm1.close();
+    realm2.close();
+  });
 }
```

Comment:
Maybe add a test, that show opening a realm with a config that has already been used to open a realm will throw?

Reply:
Scratch that.. it comes later :-)

Reply:
Yes, we have it.